File: framework/languages/java/e2e/SpotActorTransfer/Server/ActorNode/src/main/java/systems/zlink/e2e/spotactortransfer/actor/ActorNodeOptions.java
package systems.zlink.e2e.spotactortransfer.actor;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties("e2e")
public record ActorNodeOptions(
    String nodeRid, String meshEndpoint, String meshPeers, String streamEndpoint,
    String httpEndpoint, String redisLocationEndpoint, String locationKeyPrefix,
    String logDirectory, String scenario, boolean automaticTopology) {
    public ActorNodeOptions {
        required(nodeRid, "node-rid"); required(meshEndpoint, "mesh-endpoint");
        required(meshPeers, "mesh-peers"); required(streamEndpoint, "stream-endpoint");
        required(httpEndpoint, "http-endpoint"); required(redisLocationEndpoint, "redis-location-endpoint");
        required(locationKeyPrefix, "location-key-prefix"); required(logDirectory, "log-directory");
        required(scenario, "scenario");
    }
    private static void required(String value, String name) {
        if (value == null || value.isBlank()) throw new IllegalArgumentException("e2e." + name + " is required");
    }
}

File: framework/languages/java/e2e-kotlin/SpotActorTransfer/Server/ActorNode/src/main/kotlin/systems/zlink/e2e/kotlin/spotactortransfer/actor/ActorNodeOptions.kt
package systems.zlink.e2e.kotlin.spotactortransfer.actor

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("e2e")
data class ActorNodeOptions(
    val nodeRid: String,
    val meshEndpoint: String,
    val meshPeers: String,
    val streamEndpoint: String,
    val httpEndpoint: String,
    val redisLocationEndpoint: String,
    val locationKeyPrefix: String,
    val logDirectory: String,
    val scenario: String,
)
